pkg/cmd/grafana-cli/services/services.go

```diff
@@ -126,10 +127,14 @@ func RemoveInstalledPlugin(pluginPath, pluginName string) error {
 }
 
 func GetPlugin(pluginId, repoUrl string) (m.Plugin, error) {
+	logger.Debugf("getting plugin metadata from: %v pluginId: %v \n", repoUrl, pluginId)
 	body, err := sendRequest(repoUrl, "repo", pluginId)
 
 	if err != nil {
-		logger.Info("Failed to send request", "error", err)
+		logger.Info("Failed to send request", err)
+		if strings.Contains(err.Error(), "404") {
+			return m.Plugin{}, fmt.Errorf("Failed to find requested plugin, check if the plugin_id is correct. error: %v", err)
```

Comment:
Better message if plugins isn't found. I did not find a better way to test for 404 other then look at the error string :(

Reply:
The CLI is the first Go code I wrote. So it's pretty crappy :)
In this case, I think it would be better if `sendRequest` returned the `http.Response` as well as body and error. So its possible to check the StatusCode on the Response.

Reply:
Ah you are right that would be better. I looked for some status in the error and I overlooked that I could take it from response. Will change that.